qubes/storage/__init__.py

```diff
@@ -431,46 +431,44 @@ def create(self):
         os.umask(old_umask)
 
     @asyncio.coroutine
-    def clone(self, src_vm):
-        ''' Clone volumes from the specified vm '''
+    def clone_volume(self, src_vm, name):
+        ''' Clone single volume from the specified vm
+
+        :param QubesVM src_vm: source VM
+        :param str name: name of volume to clone ('root', 'private' etc)
+        :return cloned volume object
+        '''
+        config = self.vm.volume_config[name]
+        dst_pool = self.vm.app.get_pool(config['pool'])
+        dst = dst_pool.init_volume(self.vm, config)
+        src_volume = src_vm.volumes[name]
+        src_pool = src_volume.pool
+        if dst_pool == src_pool:
+            msg = "Cloning volume {!s} from vm {!s}"
+            self.vm.log.info(msg.format(src_volume.name, src_vm.name))
+            clone_op_ret = dst_pool.clone(src_volume, dst)
+        else:
+            msg = "Importing volume {!s} from vm {!s}"
+            self.vm.log.info(msg.format(src_volume.name, src_vm.name))
+            clone_op_ret = dst_pool.import_volume(
+                dst_pool, dst, src_pool, src_volume)
 
         # clone/import functions may be either synchronous or asynchronous
         # in the later case, we need to wait for them to finish
-        clone_op = {}
+        if asyncio.iscoroutine(clone_op_ret):
+            self.vm.volumes[name] = yield from clone_op_ret
+        else:
+            self.vm.volumes[name] = clone_op_ret
```

Comment:
This is very minor, but Pythonist would write:
```python
clone_op_ret = ...
if asyncio.iscoroutine(clone_op_ret):
  clone_op_ret = yield from clone_op_ret
self.vm.volumes[name] = clone_op_ret
```
That is, take advantage of the fact that Python is not static typed and we don't write the main operation (in this case `self.vm.volumes[name] = ...`) twice.